pkg/cmd/domains/create/create.go

```diff
@@ -0,0 +1,90 @@
+package create
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/MakeNowJust/heredoc"
+
+	msg "github.com/aziontech/azion-cli/messages/domains"
+	api "github.com/aziontech/azion-cli/pkg/api/domains"
+
+	"github.com/aziontech/azion-cli/utils"
+	"github.com/aziontech/azion-cli/pkg/cmdutil"
+	"github.com/spf13/cobra"
+)
+
+type Fields struct {
+	Name                 string   `json:"name"`
+	Cnames               []string `json:"cnames,omitempty"`
+	EdgeApplicationId    int      `json:"edge_application_id"`
+	DigitalCertificateId int      `json:"digital_certificate_id,omitempty"`
+	CnameAccessOnly      bool     `json:"cname_access-only,omitempty"`
+	IsActive             bool     `json:"is_active,omitempty"`
+	Path                 string   `json:"path,omitempty"`
+}
+
+func NewCmd(f *cmdutil.Factory) *cobra.Command {
+	fields := &Fields{}
+
+	cmd := &cobra.Command{
+		Use:           msg.DomainsCreateUsage,
+		Short:         msg.DomainsCreateShortDescription,
+		Long:          msg.DomainsCreateLongDescription,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		Example: heredoc.Doc(`
+        $ azioncli domains create --name asdfçlkj --cnames "asdf.com,asdfsdf.com,asdfd.com" --cname-access-only false
+        $ azioncli domains create --name withargs --application-id 1231 --active true
+        $ azioncli domains create --in "create.json"
+        `),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.Flags().Changed("in") {
+				f, err := os.ReadFile(fields.Path)
+				if err != nil {
+					return fmt.Errorf("%s %s", utils.ErrorOpeningFile, fields.Path)
+				}
+
+				err = json.Unmarshal(f, &fields)
+				if err != nil {
+					return utils.ErrorUnmarshalReader
+				}
+			}
+
+			if len(fields.Name) < 1 || fields.EdgeApplicationId < 1 {
```

Comment:
!Flags.Changed("name") || !Flags.Changed("application-id")

Reply:
what use func? if has value!

Reply:
It's the same as in other commands
```
if !cmd.Flags().Changed("active") || !cmd.Flags().Changed("code") || !cmd.Flags().Changed("name") {
					return msg.ErrorMandatoryCreateFlags
				}
```

Reply:
it looks good is what I was thinking, but there is the issue that I check the same guy twice, so I preferred doing only what is necessary, but I also don't think it's a bad thing to repeat the verification of the --in flag